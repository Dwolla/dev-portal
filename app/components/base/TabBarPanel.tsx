/* eslint-disable react/no-array-index-key */
import React, { useState } from "react";
import { Tab, Tabs, TabList, TabPanel } from "react-tabs";
import { css, Global } from "@emotion/react";
import styled from "@emotion/styled";
import classnames from "../../modules/classnames";
import { GREY_1, GREY_2, WHITE_PRIMARY, PARAGRAPH_TEXT } from "../colors";
import { ROBOTO } from "../typography";

const StyledIcon = styled.img`
  max-height: 36px;
`;

const ContentContainer = styled.div`
  padding: 30px;
  box-sizing: border-box;
  border: 1px solid ${GREY_2};
  border-radius: 0 0 5px 5px;
  border-top: unset;
  color: ${PARAGRAPH_TEXT};
  font-family: ${ROBOTO};
  font-size: 16px;
  line-height: 28px;
`;

// Props
type TabProps = {
  label: string;
  icon: string;
  iconActive: string;
  content: any;
};

type Props = {
  tabs: TabProps[];
};

function TabBarPanel({ tabs }: Props) {
  const [activeTab, setActiveTab] = useState(tabs[0]);
  const [hoverTab, setHoverTab] = useState(null);

  return (
    <>
      <Global
        styles={css`
          .container {
            background-color: ${WHITE_PRIMARY};
          }
          .tab-list {
            margin: unset;
            padding: unset;
            background-color: ${GREY_1};
            list-style: none;
            box-sizing: border-box;
            border-radius: 5px 5px 0 0;
            border-color: ${GREY_2};
            border-style: solid;
            border-width: 1px 1px 0 1px;
            display: flex;
            flex-wrap: nowrap;
            /* Tab Bar will scroll on smaller screens */
            overflow-x: auto;
            &::-webkit-scrollbar {
              display: none;
            }
          }
          .tab {
            flex-grow: 1;
            padding: 15px 20px;
            display: flex;
            align-items: center;
            justify-content: center;
            cursor: pointer;
            border-radius: 5px 5px 0 0;
            border: 1px solid ${GREY_1};
            border-bottom: 1px solid ${GREY_2};
            &.is-active {
              cursor: default;
              background-color: ${WHITE_PRIMARY};
              border-color: ${GREY_2};
              border-style: solid;
              border-width: 0 1px 0 1px;
              border-radius: 5px 5px 0 0;
              padding-top: 16px;
              :nth-of-type(1) {
                border-left: unset;
                padding-left: 21px;
              }
              :nth-last-of-type(1) {
                border-right: unset;
                padding-right: 21px;
              }
            }
          }
        `}
      ></Global>
      <Tabs
        onSelect={(index) => setActiveTab(tabs[index])}
        className="container"
      >
        <TabList className="tab-list">
          {tabs.map((tab, index) => (
            <Tab
              className={classnames(`tab`, {
                "is-active": tab === activeTab,
              })}
              key={index}
              onMouseEnter={() => setHoverTab(tab)}
              onMouseLeave={() => setHoverTab(null)}
            >
              <StyledIcon
                src={
                  tab === activeTab || tab === hoverTab
                    ? tab.iconActive
                    : tab.icon
                }
                alt={tab.label}
              />
            </Tab>
          ))}
        </TabList>
        {tabs.map((tab, index) => (
          <TabPanel key={index}>
            <ContentContainer>{tab.content}</ContentContainer>
          </TabPanel>
        ))}
      </Tabs>
    </>
  );
}

export default TabBarPanel;
