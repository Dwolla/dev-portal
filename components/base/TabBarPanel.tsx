import React, { useState } from "react";
import styled from "@emotion/styled";
import classnames from "../../modules/classnames";
import { GREY_1, GREY_2, WHITE_PRIMARY, PARAGRAPH_TEXT } from "../colors";
import { BOX_SHADOW_7 } from "../shadowDepths";
import { ROBOTO } from "../typography";

// Styles
const Container = styled.div`
  max-width: auto;
  border-radius: 5px;
  background-color: ${WHITE_PRIMARY};
  :hover {
    box-shadow: ${BOX_SHADOW_7};
  }
`;

const StyledTabBar = styled.div`
  box-sizing: border-box;
  border-radius: 5px 5px 0 0;
  border-color: ${GREY_2};
  border-style: solid;
  border-width: 1px 1px 0 1px;
  background-color: ${GREY_1};
  display: flex;
  flex-wrap: nowrap;
  /* Tab Bar will scroll on smaller screens */
  overflow-x: auto;
  &::-webkit-scrollbar {
    display: none;
  }
`;

const TabStyle = styled.div`
  flex-grow: 1;
  padding: 15px 20px;
  border-radius: 5px 5px 0 0;
  border: 1px solid ${GREY_1};
  border-bottom: 1px solid ${GREY_2};
  display: flex;
  align-items: center;
  justify-content: center;
  &.active {
    background-color: ${WHITE_PRIMARY};
    border-color: ${GREY_2};
    border-style: solid;
    border-width: 0 1px 0 1px;
    border-radius: 5px 5px 0 0;
    padding-top: 16px;
    :nth-of-type(1) {
      border-left: unset;
      padding-left: 21px;
    }
    :nth-last-of-type(1) {
      border-right: unset;
      padding-right: 21px;
    }
  }
`;

const StyledIcon = styled.img`
  max-height: 36px;
`;

const ContentContainer = styled.div`
  padding: 30px;
  box-sizing: border-box;
  border: 1px solid ${GREY_2};
  border-radius: 0 0 5px 5px;
  border-top: unset;
  color: ${PARAGRAPH_TEXT};
  font-family: ${ROBOTO};
  font-size: 16px;
  line-height: 28px;
`;

// Props
type TabProps = {
  label: string;
  icon: string;
  iconActive: string;
  content: any;
};

type Props = {
  tabs: TabProps[];
};

function TabBarPanel({ tabs }: Props) {
  const [activeTab, setActiveTab] = useState(tabs[0]);

  return (
    <Container>
      <StyledTabBar>
        {tabs.map((tab) => (
          <TabStyle
            className={classnames({ active: tab === activeTab })}
            key={tab.label}
            onClick={() => {
              setActiveTab(tab);
            }}
          >
            <StyledIcon
              src={tab === activeTab ? tab.iconActive : tab.icon}
              alt={tab.label}
            />
          </TabStyle>
        ))}
      </StyledTabBar>
      <ContentContainer>{activeTab.content}</ContentContainer>
    </Container>
  );
}

export default TabBarPanel;
