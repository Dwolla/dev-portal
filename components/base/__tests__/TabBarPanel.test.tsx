import renderer from "react-test-renderer";
import TabBarPanel from "../TabBarPanel";
import phpGray from "../../assets/images/content-icons/php-gray.svg";
import pythonGray from "../../assets/images/content-icons/python-gray.svg";
import rubyColor from "../../assets/images/content-icons/ruby-color.svg";
import rubyGray from "../../assets/images/content-icons/ruby-gray.svg";
import cSharpGray from "../../assets/images/content-icons/c-sharp-gray.svg";
import kotlinGray from "../../assets/images/content-icons/kotlin-gray.svg";
import nodeGray from "../../assets/images/content-icons/node-gray.svg";

const content = [
  {
    label: "node",
    icon: nodeGray,
    iconActive: nodeGray,
    content: "Here's some content related to Node.",
  },
  {
    label: "ruby",
    icon: rubyGray,
    iconActive: rubyColor,
    content: "Here's some content related to Ruby.",
  },
  {
    label: "python",
    icon: pythonGray,
    iconActive: pythonGray,
    content: "Here's some content related to Python.",
  },
  {
    label: "php",
    icon: phpGray,
    iconActive: phpGray,
    content: "Here's some content related to PHP.",
  },
  {
    label: "c-sharp",
    icon: cSharpGray,
    iconActive: cSharpGray,
    content: "Here's some content related to C Sharp.",
  },
  {
    label: "kotlin",
    icon: kotlinGray,
    iconActive: kotlinGray,
    content: "Here's some content related to Kotlin.",
  },
];

test("TabBarPanel", () => {
  const tree = renderer.create(<TabBarPanel tabs={content} />).toJSON();

  expect(tree).toMatchSnapshot();
});
