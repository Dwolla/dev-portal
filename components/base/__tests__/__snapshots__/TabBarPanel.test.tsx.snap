// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabBarPanel 1`] = `
<div
  className="container"
  data-tabs={true}
  onClick={[Function]}
  onKeyDown={[Function]}
>
  <ul
    className="tab-list"
    role="tablist"
  >
    <li
      aria-controls="react-tabs-1"
      aria-disabled="false"
      aria-selected="true"
      className="tab is-active react-tabs__tab--selected"
      id="react-tabs-0"
      role="tab"
      tabIndex="0"
    >
      <img
        alt="node"
        className="css-rh2ze1-StyledIcon efzvwdu0"
        src="test-image-stub"
      />
    </li>
    <li
      aria-controls="react-tabs-3"
      aria-disabled="false"
      aria-selected="false"
      className="tab"
      id="react-tabs-2"
      role="tab"
      tabIndex={null}
    >
      <img
        alt="ruby"
        className="css-rh2ze1-StyledIcon efzvwdu0"
        src="test-image-stub"
      />
    </li>
    <li
      aria-controls="react-tabs-5"
      aria-disabled="false"
      aria-selected="false"
      className="tab"
      id="react-tabs-4"
      role="tab"
      tabIndex={null}
    >
      <img
        alt="python"
        className="css-rh2ze1-StyledIcon efzvwdu0"
        src="test-image-stub"
      />
    </li>
    <li
      aria-controls="react-tabs-7"
      aria-disabled="false"
      aria-selected="false"
      className="tab"
      id="react-tabs-6"
      role="tab"
      tabIndex={null}
    >
      <img
        alt="php"
        className="css-rh2ze1-StyledIcon efzvwdu0"
        src="test-image-stub"
      />
    </li>
    <li
      aria-controls="react-tabs-9"
      aria-disabled="false"
      aria-selected="false"
      className="tab"
      id="react-tabs-8"
      role="tab"
      tabIndex={null}
    >
      <img
        alt="c-sharp"
        className="css-rh2ze1-StyledIcon efzvwdu0"
        src="test-image-stub"
      />
    </li>
    <li
      aria-controls="react-tabs-11"
      aria-disabled="false"
      aria-selected="false"
      className="tab"
      id="react-tabs-10"
      role="tab"
      tabIndex={null}
    >
      <img
        alt="kotlin"
        className="css-rh2ze1-StyledIcon efzvwdu0"
        src="test-image-stub"
      />
    </li>
  </ul>
  <div
    aria-labelledby="react-tabs-0"
    className="react-tabs__tab-panel react-tabs__tab-panel--selected"
    id="react-tabs-1"
    role="tabpanel"
  >
    <div
      className="css-2vq36g-ContentContainer efzvwdu1"
    >
      Here's some content related to Node.
    </div>
  </div>
  <div
    aria-labelledby="react-tabs-2"
    className="react-tabs__tab-panel"
    id="react-tabs-3"
    role="tabpanel"
  />
  <div
    aria-labelledby="react-tabs-4"
    className="react-tabs__tab-panel"
    id="react-tabs-5"
    role="tabpanel"
  />
  <div
    aria-labelledby="react-tabs-6"
    className="react-tabs__tab-panel"
    id="react-tabs-7"
    role="tabpanel"
  />
  <div
    aria-labelledby="react-tabs-8"
    className="react-tabs__tab-panel"
    id="react-tabs-9"
    role="tabpanel"
  />
  <div
    aria-labelledby="react-tabs-10"
    className="react-tabs__tab-panel"
    id="react-tabs-11"
    role="tabpanel"
  />
</div>
`;
